halo/app/ready.go

```diff
@@ -0,0 +1,75 @@
+package app
+
+import (
+	"encoding/json"
+	"io"
+	"sync"
+
+	"github.com/omni-network/omni/lib/errors"
+)
+
+// Consist of all flags describing the current readiness status of a halo node.
+type ReadyResponse struct {
```

Comment:
don't export types/funcs by default, rather make it unexported. Same as the fields below, since they are accessed via methods, making them private is better since it makes it clear that callers/users shoudln't read/write fields directly.

Reply:
mmm, but this will result in it not seriazling the json, so maybe fields can be exported. Just add a godoc that fields shouldn't be written/read directly. If you want to be purist, you could have a `readyResponseJSON` that is only used for serialization.

Reply:
My plan was also to import the type in the test to unmarshall the endpoint output. I think this will not work for unexported type, right? Would you prefer me to just redefine the struct in the test instead?

Reply:
yeah, don't export just for tests, rather make the test `_internal` or duplicate the struct in the tests